x-pack/plugins/ml/public/application/services/ml_api_service/index.ts

```diff
@@ -53,6 +53,11 @@ import { savedObjectsApiProvider } from './saved_objects';
 import { trainedModelsApiProvider } from './trained_models';
 import { notificationsProvider } from './notifications';
 
+export interface ExtendedSecurityHasPrivilegesResponse
+  extends estypes.SecurityHasPrivilegesResponse {
+  securityDisabled?: boolean;
```

Comment:
this should also contain `upgradeInProgress: boolean`

Reply:
I think these this type is a wrong. looking at the endpoint, it'll return either
```
{
  securityDisabled: true,
  upgradeInProgress,
}
```
or `SecurityHasPrivilegesResponse` with `upgradeInProgress` added in.

Reply:
Updated in 99b23749c53de256e92c2622795e6f319e328e7c

`securityDisabled` is only added in one of the responses so needs to be in but good catch on adding in `upgradeInProgress` 🙏

Reply:
>So I think extending `estypes.SecurityHasPrivilegesResponse` is incorrect.

Sorry, I wasn't clear in what I mean here.
I mean that we should probably have a union of either responses.